src/utils/atomicSwap.js

```diff
@@ -340,3 +333,69 @@ export const updatePersistentStorage = (proposalList) => {
 
     walletUtil.setListenedProposals(simplifiedStorage);
 }
+
+/**
+ * Calculates the object that will be stored in Redux, containing all the helper values for a more
+ * enriched exhibition on the Atomic Swap screens
+ * @param {string} proposalId
+ * @param {string} password
+ * @param {string} partialTx
+ * @param {HathorWallet} wallet
+ * @param {string} [signatures]
+ * @return {ReduxProposalData}
+ */
+export const generateReduxObjFromProposal = (proposalId, password, partialTx, wallet, signatures) => {
+    /** @type ReduxProposalData */
+    const rObj = {
+        id: proposalId,
+        password: password,
+        status: PROPOSAL_DOWNLOAD_STATUS.INVALIDATED, // We do not know the current state from the calculations
+        data: { // Only the minimum information. Any other data already retrieved should be added afterward
+            id: proposalId,
+            partialTx,
+            history: [],
+        },
+        updatedAt: new Date().valueOf(),
+    };
+    // Building the object to retrieve data from
+    const txProposal = new PartialTxProposal.fromPartialTx(partialTx, wallet.getNetworkObject());
+
+    // Calculating the amount of tokens
+    const balance = txProposal.calculateBalance(wallet);
+    rObj.data.amountTokens = Object.keys(balance).length;
+
+    // Calculating signature status
+    let sigStatus;
+    const amountInputs = txProposal.partialTx.getTx().inputs.length;
+    const amountSigs = signatures
+      ? signatures.split('|').length - 2 // Removing the prefix and txHex, sigs remain
+      : 0;
+    if (amountSigs < 1) {
+        sigStatus = PROPOSAL_SIGNATURE_STATUS.OPEN;
+    } else if (amountSigs < amountInputs) {
+        sigStatus = PROPOSAL_SIGNATURE_STATUS.PARTIALLY_SIGNED;
+    } else {
+        sigStatus = PROPOSAL_SIGNATURE_STATUS.SIGNED;
+    }
+    rObj.data.signatureStatus = sigStatus;
+
+    return rObj;
+}
+
+/**
+ * Returns the Atomic Swap Service base URL,
+ * @param {string} network Network name for fetching the default base server url
```

Comment:
Added on [7b3d362](https://github.com/HathorNetwork/hathor-wallet/pull/400/commits/7b3d36219cb382d33b17ac5fd8883cbb1a91e731)

Reply:
Is the docstring correct? I thought we were returning the URL, that's why I asked for the return type.

Reply:
Yes, the method retrieves the URL internally and uses it to set the lib base url. But never returns it to the caller, as it would have no use in that context.